src/OfflineWorkflowsSample/OfflineWorkflowsSample/Converters/ItemToImageSourceConverter.cs

```diff
@@ -0,0 +1,42 @@
+﻿using Esri.ArcGISRuntime.Portal;
+using Esri.ArcGISRuntime.UI;
+using System;
+using Windows.UI.Xaml.Data;
+using Windows.UI.Xaml.Media.Imaging;
+using Esri.ArcGISRuntime;
+
+namespace OfflineWorkflowSample.Converters
+{
+    /// <summary>
+    /// Converts from an Item (PortalItem or LocalItem) to an image. This is needed
+    /// because the thumbnail could be an image or a link to an image.
+    /// </summary>
+    class ItemToImageSourceConverter : IValueConverter
+    {
+        public object Convert(object value, Type targetType, object parameter, string language)
+        {
+            if (value is Item mapItem)
+            {
+                if (mapItem.ThumbnailUri != null)
+                {
+                    // Sometimes image URIs have a . appended to them... BitmapImage doesn't like that.
+                    return new BitmapImage(new Uri(mapItem.ThumbnailUri.OriginalString.TrimEnd('.')));
+                }
+
+                if (mapItem.Thumbnail != null &&
+                    mapItem.Thumbnail.LoadStatus == LoadStatus.Loaded &&
+                    mapItem.Thumbnail.Width > 0)
+                {
+                    return mapItem.Thumbnail.ToImageSourceAsync().Result;
```

Comment:
It is true that UWP does not provide a nice mechanism for "async conversion," but blocking for ToImageSourceAsync is not okay. This could be a case for wrapping Item in a ViewModel, which can convert from model representation (ThumbnailUri / Thumbnail / fallback image) to View representation (just an ImageSource), and even do so lazily/asynchronously by firing a PCE when it's ready.